SampleGallery/Samples/SDK 16299/LinearGradients/LinearGradients.xaml.cs

```diff
@@ -0,0 +1,263 @@
+﻿//*********************************************************
+//
+// Copyright (c) Microsoft. All rights reserved.
+// This code is licensed under the MIT License (MIT).
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, 
+// INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, 
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. 
+// IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, 
+// DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, 
+// TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH 
+// THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+//
+//*********************************************************
+
+using System;
+using Windows.UI.Xaml;
+using Windows.UI.Xaml.Input;
+using Windows.UI.Composition;
+using Windows.UI.Xaml.Hosting;
+using Windows.UI;
+using System.Numerics;
+
+namespace CompositionSampleGallery
+{
+    public sealed partial class LinearGradients : SamplePage
+    {
+        public LinearGradients()
+        {
+            this.InitializeComponent();
+        }
+
+        public static string StaticSampleName => "Linear Gradients";
+        public override string SampleName => StaticSampleName;
+        public static string StaticSampleDescription => "Simple visual effect showing linear gradient animations. The first and last words will flip the colors from left to right.Clicking on any of the words will change the color scheme from warm to cool and back when clicked again." + Environment.NewLine + "This sample is to show how linear gradients can highlight content in colorful ways.";
+        public override string SampleDescription => StaticSampleDescription;
+
+        private Compositor _compositor;
+
+        private SpriteVisual _vis1;
+        private SpriteVisual _vis2;
+        private SpriteVisual _vis3;
+        private SpriteVisual _vis4;
+
+        private CompositionLinearGradientBrush _brush1;
+        private CompositionColorGradientStop _b1GradientStop1;
+        private CompositionColorGradientStop _b1GradientStop2;
+
+
+        private Vector3KeyFrameAnimation _scaleAnim;
+
+        private ColorKeyFrameAnimation _changeStopToHoneydew;
+        private ColorKeyFrameAnimation _changeStopToDeepPink;
+        private ColorKeyFrameAnimation _changeStopToLightSkyBlue;
+        private ColorKeyFrameAnimation _changeStopToTeal;
+
+        private Color _deepPink;
+        private Color _honeydew;
+        private Color _lightSkyBlue;
+        private Color _teal;
+
+        private void Page_Loaded(object sender, RoutedEventArgs e)
```

Comment:
The contructor would be a better place to do this work. It will only get called once while Page_Loaded can get called multiple times. Also the constructor will allow most of your fields to be readonly.

Reply:
is this referencing the compositor being initialized in page_loaded?

Reply:
i don't really know what this comment means